contracts/src/gateway/router/XnetMessagingFacet.sol

```diff
@@ -29,4 +29,28 @@ contract XnetMessagingFacet is GatewayActorModifiers {
     function applyCrossMessages(IpcEnvelope[] calldata crossMsgs) external systemActorOnly {
         LibGateway.applyMessages(s.networkName.getParentSubnet(), crossMsgs);
     }
+
+    /// @notice Applies top-down cross-net message locally one by one.
```

Comment:
This description does not tell what `ret` is but `ret` is in the name of the method.